include/boost/beast/http/impl/basic_parser.ipp

```diff
@@ -786,30 +784,46 @@ do_field(field f,
     // Content-Length
     if(f == field::content_length)
     {
-        if(f_ & flagContentLength)
+        auto bad_content_length = [&ec]
         {
-            // duplicate
             ec = error::bad_content_length;
-            return;
-        }
+        };
 
+        // conflicting field
         if(f_ & flagChunked)
+            return bad_content_length();
+
+        // Content-length may be a comma-separated list of integers
+        auto tokens_unprocessed = 1 +
+            std::count(value.begin(), value.end(), ',');
+        auto tokens = opt_token_list(value);
+        if (tokens.begin() == tokens.end() ||
+            !validate_list(tokens))
+                return bad_content_length();
+
+        auto existing = this->impl_content_length();
+        for (auto tok : tokens)
         {
-            // conflicting field
-            ec = error::bad_content_length;
-            return;
+            std::uint64_t v;
+            if (!parse_dec(tok, v))
+                return bad_content_length();
+            --tokens_unprocessed;
+            if (existing.has_value())
+            {
+                if (v != *existing)
+                    return bad_content_length();
+            }
+            else
+                existing = v;
```

Comment:
nit: if the `if` has braces then the `else` needs them too

Reply:
ok